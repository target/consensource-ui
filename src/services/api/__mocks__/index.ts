export * from './factory';
export * from './certificate';
