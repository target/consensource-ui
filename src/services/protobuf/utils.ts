import {
  CertificateRegistryPayload,
  ICertificateRegistryPayload,
} from './compiled';

export interface PayloadInfo {
  payloadBytes: string | Buffer | NodeJS.TypedArray | DataView;
  inputs: string[];
  outputs: string[];
}

export const ACTIONS = CertificateRegistryPayload.Action;

/**
 * Encodes a CertificateRegistryPayload message
 * @param message CertificateRegistryPayload message or plain object to encode
 */
export function encodePayload(
  payload: ICertificateRegistryPayload,
): Uint8Array {
  return CertificateRegistryPayload.encode(payload).finish();
}
