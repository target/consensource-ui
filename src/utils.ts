// Util collection

/**
 * Selects keys from an object
 */
export const pluck = (obj: any, ...keys: Array<any>): any =>
    keys.reduce((o, k) => {
        if (typeof obj[k] !== 'undefined') {
            o[k] = obj[k];
        }
        return o;
    }, {});

/**
 * Merges objects, starting from left to right
 */
