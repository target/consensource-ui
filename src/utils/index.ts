export function getLocaleFromUnix(timestampSeconds: number) {
  const timestampMS = timestampSeconds * 1000;
  return new Date(timestampMS).toLocaleDateString();
}
