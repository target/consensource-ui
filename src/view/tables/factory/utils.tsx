import { MUIDataTableOptions, MUIDataTableChip } from 'mui-datatables';

export const DEFAULT_ROWS_PER_PAGE = 100;

export const textLabels: MUIDataTableOptions['textLabels'] = {
  body: {
    noMatch: 'No factories found',
  },
};

/**
 * Default props for the filter chips displayed in the
 * `<TableFilterList />`.
 */
export const filterChipProps: MUIDataTableChip = {
  color: 'secondary',
  variant: 'default',
};

/**
 * Utility function to convert a string to an array, or return
 * the original param if an array was passed.
 *
 * This operation is frequent when decoding URL query params.
 */
export const convertStrToArray = (val: string[] | string) => {
  return typeof val === 'string' ? [val] : val;
};
