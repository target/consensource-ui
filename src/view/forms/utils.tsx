import React from 'react';
import { Grid, Typography, TextFieldProps } from '@material-ui/core';
import { BatchStatusRes } from 'services/api';

export interface TransactionFormProps {
  setBatchStatusLink: (statusLink: BatchStatusRes['link']) => void;
}

export type onChangeEvent = Parameters<
  NonNullable<TextFieldProps['onChange']>
>[0];

/**
 * Helper function that checks if any values in `state` have not been set.
 * Returns true if there is any key in `state` that is falsey.
 */
export function hasEmptyFields<T>(state: T) {
  return Object.keys(state).some((key) => !state[key as keyof T]);
}

export interface FormErrMsgProps {
  msg?: string;
}

export function FormErrMsg({ msg }: FormErrMsgProps) {
  if (!msg) {
    return null;
  }

  return (
    <Grid item>
      <Typography variant="caption" color="error">
        {msg}
      </Typography>
    </Grid>
  );
}
