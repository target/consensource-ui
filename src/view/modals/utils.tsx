import { DialogProps as MUIDialogProps } from '@material-ui/core';
import { FactoryResData } from 'services/api';

export interface DialogProps extends MUIDialogProps {
  handleClose: () => void;
}

export interface OrgDialogProps extends DialogProps {
  existing_org: FactoryResData;
}
