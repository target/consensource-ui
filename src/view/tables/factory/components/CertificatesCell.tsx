import React from 'react';
import { CertResData } from 'services/api';
import {
  List,
  ListItem,
  Tooltip,
  makeStyles,
  createStyles,
} from '@material-ui/core';
import { UnstyledLink } from 'view/components';

export interface CertificatesCellProps {
  // TODO: Remove optional `certificates` once we fix backend issues
  certificates?: CertResData[];
}

const useStyles = makeStyles(({ palette }) =>
  createStyles({
    link: {
      textDecoration: 'none',
      color: 'inherit',
    },
    certName: {
      borderBottom: `2px solid ${palette.primary.light}`,
      cursor: 'pointer',
      '&:hover': {
        borderBottom: `2px solid ${palette.primary.main}`,
      },
    },
  }),
);

export const CertificatesCell = ({ certificates }: CertificatesCellProps) => {
  const classes = useStyles();

  if (!certificates || certificates.length === 0) {
    return (
      <ListItem>
        <i>None</i>
      </ListItem>
    );
  }

  return (
    <List>
      {certificates.map(({ standard_name, id, valid_to }) => (
        <UnstyledLink to={`certifications/${id}`}>
          <ListItem key={id}>
            <Tooltip
              placement="top"
              title={`Valid until ${new Date(valid_to).toLocaleDateString()}`}
            >
              <i className={classes.certName}>{standard_name}</i>
            </Tooltip>
          </ListItem>
        </UnstyledLink>
      ))}
    </List>
  );
};
