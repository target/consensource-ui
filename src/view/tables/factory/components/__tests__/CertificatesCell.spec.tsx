import React from 'react';
import { render, screen } from 'utils/testing';
import { mockCerts } from 'services/api/__mocks__';
import userEvent from '@testing-library/user-event';
import { getLocaleFromUnix } from 'utils';
import { CertificatesCell } from '../CertificatesCell';

describe('<CertificatesCell />', () => {
  it('renders without any certs', () => {
    const { container } = render(<CertificatesCell certificates={[]} />);
    expect(screen.getByText('None')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  describe('with certs', () => {
    const { data: certs } = mockCerts;

    it('renders', () => {
      const { container } = render(<CertificatesCell certificates={certs} />);
      expect(container).toMatchSnapshot();
    });

    it('sets the link destination', () => {
      const { id } = certs[0];

      render(<CertificatesCell certificates={certs} />);

      expect(screen.getByRole('link')).toHaveAttribute(
        'href',
        `/certifications/${id}`,
      );
    });

    it('uses the valid_to field in the tooltip', async () => {
      const { valid_to } = certs[0];
      const data = getLocaleFromUnix(valid_to);

      render(<CertificatesCell certificates={certs} />);

      userEvent.hover(screen.getByTestId('cert-cell-tooltip'));

      expect(
        await screen.findByText(`Valid until ${data}`),
      ).toBeInTheDocument();
    });
  });
});
