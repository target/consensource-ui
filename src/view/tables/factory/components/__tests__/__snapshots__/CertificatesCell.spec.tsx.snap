// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CertificatesCell /> renders without any certs 1`] = `
<div>
  <li
    class="MuiListItem-root MuiListItem-gutters"
  >
    <i>
      None
    </i>
  </li>
</div>
`;

exports[`<CertificatesCell /> with certs renders 1`] = `
<div>
  <ul
    class="MuiList-root MuiList-padding"
  >
    <a
      class="makeStyles-link"
      href="certifications/a978112ca1bbdcafac231b39a23dc4da786eff8147c4e72b9807785afee4811"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <i
          class="makeStyles-certName"
          data-testid="cert-cell-tooltip"
          title="Valid until 8/18/2020"
        >
          Test Standard
        </i>
      </li>
    </a>
  </ul>
</div>
`;
